src/components/instant.rs

```diff
@@ -38,17 +78,15 @@ impl Instant {
     ///
     /// Temporal-Proposal equivalent: `AddDurationToOrSubtractDurationFrom`.
     pub(crate) fn add_to_instant(&self, duration: &TimeDuration) -> TemporalResult<Self> {
-        let result = self.epoch_nanoseconds()
+        let current_nanos = self.epoch_nanoseconds() as f64;
```

Comment:
Why is this cast as an `f64` here?

Reply:
I think this can go either way. It was mostly meant to limit the amount of casts since the addition with the duration fields would require casting at f64 fields to i128.